synapse/handlers/room.py

```diff
@@ -752,6 +752,15 @@ async def create_room(
         if not allowed_by_third_party_rules:
             raise SynapseError(403, "Room visibility value not allowed.")
 
+        if is_public:
+            if not self.config.roomdirectory.is_publishing_room_allowed(
+                user_id, room_id, room_alias
+            ):
+                # Lets just return a generic message, as there may be all sorts of
```

Comment:
Not your typo, but while we're here:

```suggestion
                # Let's just return a generic message, as there may be all sorts of
```